extensions/sql-migration/src/wizard/createMigrationControllerDialog.ts

```diff
@@ -0,0 +1,485 @@
+		const errors: string[] = [];
+		if (!subscription) {
+			errors.push(constants.INVALID_SUBSCRIPTION_ERROR);
+		}
+		if (!resourceGroup) {
+			errors.push(constants.INVALID_RESOURCE_GROUP_ERROR);
+		}
+		if (!region) {
+			errors.push(constants.INVALID_REGION_ERROR);
+		}
+		if (!controllerName || controllerName.length === 0) {
+			errors.push(constants.INVALID_CONTROLLER_NAME_ERROR);
+		}
+		return errors.join(os.EOL);
+	}
+
+	private async populateSubscriptions(): Promise<void> {
+		this.migrationControllerSubscriptionDropdown.loading = true;
+		this.migrationControllerResourceGroupDropdown.loading = true;
+		const subscriptions = await getSubscriptions(this.migrationStateModel.azureAccount);
+
+		let subscriptionDropdownValues: azdata.CategoryValue[] = [];
+		if (subscriptions && subscriptions.length > 0) {
+
+			subscriptions.forEach((subscription) => {
+				this._subscriptionMap.set(subscription.id, subscription);
+				subscriptionDropdownValues.push({
+					name: subscription.id,
+					displayName: subscription.name + ' - ' + subscription.id,
+				});
+			});
+
+
+		} else {
+			subscriptionDropdownValues = [
+				{
+					displayName: constants.NO_SUBSCRIPTIONS_FOUND,
+					name: ''
+				}
+			];
+		}
+
+		this.migrationControllerSubscriptionDropdown.values = subscriptionDropdownValues;
+		this.migrationControllerSubscriptionDropdown.loading = false;
+		this.populateResourceGroups();
+	}
+
+	private async populateResourceGroups(): Promise<void> {
+		this.migrationControllerResourceGroupDropdown.loading = true;
+		let subscription = this._subscriptionMap.get((this.migrationControllerSubscriptionDropdown.value as azdata.CategoryValue).name)!;
+		const resourceGroups = await getResourceGroups(this.migrationStateModel.azureAccount, subscription);
+		let resourceGroupDropdownValues: azdata.CategoryValue[] = [];
+		if (resourceGroups && resourceGroups.length > 0) {
+			resourceGroups.forEach((resourceGroup) => {
+				resourceGroupDropdownValues.push({
+					name: resourceGroup.name,
+					displayName: resourceGroup.name
+				});
+			});
+		} else {
+			resourceGroupDropdownValues = [
+				{
+					displayName: constants.RESOURCE_GROUP_NOT_FOUND,
+					name: ''
+				}
+			];
+		}
+		this.migrationControllerResourceGroupDropdown.values = resourceGroupDropdownValues;
+		this.migrationControllerResourceGroupDropdown.loading = false;
+	}
+
+	private createControllerStatus(): azdata.FlexContainer {
+
+		const informationTextBox = this._view.modelBuilder.text().withProps({
+			value: constants.CONTROLLER_DIALOG_CONTROLLER_CONTAINER_DESCRIPTION
+		}).component();
+
+		const expressSetupTitle = this._view.modelBuilder.text().withProps({
+			value: constants.CONTROLLER_OPTION1_HEADING,
+			CSSStyles: {
+				'font-weight': 'bold'
+			}
+		}).component();
+
+		const expressSetupLink = this._view.modelBuilder.hyperlink().withProps({
+			label: constants.CONTROLLER_OPTION1_SETUP_LINK_TEXT,
+			url: ''
```

Comment:
Filling this in later?

Reply:
Yup, right now a notification is shown that says feature not available.